kore/src/Kore/Step/Rule/Combine.hs

```diff
@@ -176,7 +179,17 @@ mergeRules
     => NonEmpty (RewriteRule variable)
     -> simplifier [RewriteRule variable]
 mergeRules (a :| []) = return [a]
-mergeRules rules = BranchT.gather $ do
+mergeRules rules =
```

Comment:
This is up to you, but you could use a view pattern here, get rid of the helper (`mergeDisjointVarRules`), and that would remove the possibility of anyone accidentally using the helper wrong in the future.

```suggestion
mergeRules (renameRulesVariables . Foldable.toList -> rules) =
```